# **CIRRHOSIS**
It is a chronic liver disease marked by degeneration of cells, inflammation, and fibrous thickening of tissue. It is typically a result of alcoholism or hepatitis.

# Study Design
A total of 424 PBC(primary biliary cirrhosis) patients, referred to Mayo Clinic during that ten-year interval, met eligibility criteria for the **randomized placebo-controlled trial of the drug D-penicillamine**.

***Variable Information***

1.   ID: unique identifier
2.   N_Days: number of days between registration and the earlier of death, transplantation, or study analysis time in July 1986
3.  Status: status of the patient C (censored), CL (censored due to liver tx), or D (death)
4. Drug: type of drug D-penicillamine or placebo
5. Age: age in [days]
6. Sex: M (male) or F (female)
7. Ascites: presence of ascites N (No) or Y (Yes)
8. Hepatomegaly: presence of hepatomegaly N (No) or Y (Yes)
9. Spiders: presence of spiders N (No) or Y (Yes)
10. Edema: presence of edema N (no edema and no diuretic therapy for edema), S (edema present without diuretics, or edema resolved by diuretics), or Y (edema despite diuretic therapy)
11. Bilirubin: serum bilirubin in [mg/dl]
12. Cholesterol: serum cholesterol in [mg/dl]
13. Albumin: albumin in [gm/dl]
14. Copper: urine copper in [ug/day]
15. Alk_Phos: alkaline phosphatase in [U/liter]
16. SGOT: SGOT in [U/ml]
17. Triglycerides: triglicerides in [mg/dl]
18. Platelets: platelets per cubic [ml/1000]
19. Prothrombin: prothrombin time in seconds [s]
20. Stage: histologic stage of disease (1, 2, 3, or 4)

***Acknowledgement***

This dataset is taken from Kaggle.com.

**Cirrhosis Prediction Dataset**

https://www.kaggle.com/datasets/fedesoriano/cirrhosis-prediction-dataset

In [92]:
import pandas as pd

In [93]:
from google.colab import files
uploaded = files.upload()
df = pd.read_csv("cirrhosis.csv")
df.head()

Saving cirrhosis.csv to cirrhosis (3).csv


ID  N_Days Status             Drug    Age Sex Ascites Hepatomegaly Spiders  \
0   1     400      D  D-penicillamine  21464   F       Y            Y       Y   
1   2    4500      C  D-penicillamine  20617   F       N            Y       Y   
2   3    1012      D  D-penicillamine  25594   M       N            N       N   
3   4    1925      D  D-penicillamine  19994   F       N            Y       Y   
4   5    1504     CL          Placebo  13918   F       N            Y       Y   

  Edema  Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos    SGOT  \
0     Y       14.5        261.0     2.60   156.0    1718.0  137.95   
1     N        1.1        302.0     4.14    54.0    7394.8  113.52   
2     S        1.4        176.0     3.48   210.0     516.0   96.10   
3     S        1.8        244.0     2.54    64.0    6121.8   60.63   
4     N        3.4        279.0     3.53   143.0     671.0  113.15   

   Tryglicerides  Platelets  Prothrombin  Stage  
0          172.0      190.0         12.2    4.0  
1           88.0      221.0         10.6    3.0  
2           55.0      151.0         12.0    4.0  
3           92.0      183.0         10.3    4.0  
4           72.0      136.0         10.9    3.0

In [94]:
df.tail()

ID  N_Days Status Drug    Age Sex Ascites Hepatomegaly Spiders Edema  \
413  414     681      D  NaN  24472   F     NaN          NaN     NaN     N   
414  415    1103      C  NaN  14245   F     NaN          NaN     NaN     N   
415  416    1055      C  NaN  20819   F     NaN          NaN     NaN     N   
416  417     691      C  NaN  21185   F     NaN          NaN     NaN     N   
417  418     976      C  NaN  19358   F     NaN          NaN     NaN     N   

     Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos  SGOT  Tryglicerides  \
413        1.2          NaN     2.96     NaN       NaN   NaN            NaN   
414        0.9          NaN     3.83     NaN       NaN   NaN            NaN   
415        1.6          NaN     3.42     NaN       NaN   NaN            NaN   
416        0.8          NaN     3.75     NaN       NaN   NaN            NaN   
417        0.7          NaN     3.29     NaN       NaN   NaN            NaN   

     Platelets  Prothrombin  Stage  
413      174.0         10.9    3.0  
414      180.0         11.2    4.0  
415      143.0          9.9    3.0  
416      269.0         10.4    3.0  
417      350.0         10.6    4.0

In [95]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 20 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   ID             418 non-null    int64  
 1   N_Days         418 non-null    int64  
 2   Status         418 non-null    object 
 3   Drug           312 non-null    object 
 4   Age            418 non-null    int64  
 5   Sex            418 non-null    object 
 6   Ascites        312 non-null    object 
 7   Hepatomegaly   312 non-null    object 
 8   Spiders        312 non-null    object 
 9   Edema          418 non-null    object 
 10  Bilirubin      418 non-null    float64
 11  Cholesterol    284 non-null    float64
 12  Albumin        418 non-null    float64
 13  Copper         310 non-null    float64
 14  Alk_Phos       312 non-null    float64
 15  SGOT           312 non-null    float64
 16  Tryglicerides  282 non-null    float64
 17  Platelets      407 non-null    float64
 18  Prothrombi

In [96]:
df.shape

(418, 20)

In [97]:
df.describe()

ID       N_Days           Age   Bilirubin  Cholesterol  \
count  418.000000   418.000000    418.000000  418.000000   284.000000   
mean   209.500000  1917.782297  18533.351675    3.220813   369.510563   
std    120.810458  1104.672992   3815.845055    4.407506   231.944545   
min      1.000000    41.000000   9598.000000    0.300000   120.000000   
25%    105.250000  1092.750000  15644.500000    0.800000   249.500000   
50%    209.500000  1730.000000  18628.000000    1.400000   309.500000   
75%    313.750000  2613.500000  21272.500000    3.400000   400.000000   
max    418.000000  4795.000000  28650.000000   28.000000  1775.000000   

          Albumin      Copper      Alk_Phos        SGOT  Tryglicerides  \
count  418.000000  310.000000    312.000000  312.000000     282.000000   
mean     3.497440   97.648387   1982.655769  122.556346     124.702128   
std      0.424972   85.613920   2140.388824   56.699525      65.148639   
min      1.960000    4.000000    289.000000   26.350000      33.000000   
25%      3.242500   41.250000    871.500000   80.600000      84.250000   
50%      3.530000   73.000000   1259.000000  114.700000     108.000000   
75%      3.770000  123.000000   1980.000000  151.900000     151.000000   
max      4.640000  588.000000  13862.400000  457.250000     598.000000   

        Platelets  Prothrombin       Stage  
count  407.000000   416.000000  412.000000  
mean   257.024570    10.731731    3.024272  
std     98.325585     1.022000    0.882042  
min     62.000000     9.000000    1.000000  
25%    188.500000    10.000000    2.000000  
50%    251.000000    10.600000    3.000000  
75%    318.000000    11.100000    4.000000  
max    721.000000    18.000000    4.000000

In [98]:
df.nunique()

ID               418
N_Days           399
Status             3
Drug               2
Age              344
Sex                2
Ascites            2
Hepatomegaly       2
Spiders            2
Edema              3
Bilirubin         98
Cholesterol      201
Albumin          154
Copper           158
Alk_Phos         295
SGOT             179
Tryglicerides    146
Platelets        243
Prothrombin       48
Stage              4
dtype: int64

In [99]:
df.columns

Index(['ID', 'N_Days', 'Status', 'Drug', 'Age', 'Sex', 'Ascites',
       'Hepatomegaly', 'Spiders', 'Edema', 'Bilirubin', 'Cholesterol',
       'Albumin', 'Copper', 'Alk_Phos', 'SGOT', 'Tryglicerides', 'Platelets',
       'Prothrombin', 'Stage'],
      dtype='object')

In [100]:
df.corr()

ID    N_Days       Age  Bilirubin  Cholesterol   Albumin  \
ID             1.000000 -0.354305  0.037136  -0.062154     0.032897 -0.128924   
N_Days        -0.354305  1.000000 -0.125934  -0.403953    -0.138236  0.430829   
Age            0.037136 -0.125934  1.000000   0.002362    -0.157620 -0.182350   
Bilirubin     -0.062154 -0.403953  0.002362   1.000000     0.397129 -0.314177   
Cholesterol    0.032897 -0.138236 -0.157620   0.397129     1.000000 -0.069733   
Albumin       -0.128924  0.430829 -0.182350  -0.314177    -0.069733  1.000000   
Copper        -0.098663 -0.364809  0.061549   0.456918     0.126115 -0.264771   
Alk_Phos      -0.352856  0.149269 -0.047247   0.116984     0.149473 -0.101456   
SGOT          -0.012097 -0.225492 -0.149869   0.441730     0.353246 -0.220047   
Tryglicerides -0.034100 -0.153000  0.022065   0.436748     0.276830 -0.103417   
Platelets     -0.076699  0.151361 -0.148201  -0.013435     0.191710  0.158659   
Prothrombin   -0.191930 -0.111470  0.113760   0.314894    -0.030811 -0.200592   
Stage         -0.033757 -0.366193  0.189083   0.200731     0.011164 -0.305296   

                 Copper  Alk_Phos      SGOT  Tryglicerides  Platelets  \
ID            -0.098663 -0.352856 -0.012097      -0.034100  -0.076699   
N_Days        -0.364809  0.149269 -0.225492      -0.153000   0.151361   
Age            0.061549 -0.047247 -0.149869       0.022065  -0.148201   
Bilirubin      0.456918  0.116984  0.441730       0.436748  -0.013435   
Cholesterol    0.126115  0.149473  0.353246       0.276830   0.191710   
Albumin       -0.264771 -0.101456 -0.220047      -0.103417   0.158659   
Copper         1.000000  0.187357  0.293829       0.279852  -0.064403   
Alk_Phos       0.187357  1.000000  0.112217       0.180082   0.143733   
SGOT           0.293829  0.112217  1.000000       0.126119  -0.120147   
Tryglicerides  0.279852  0.180082  0.126119       1.000000   0.103212   
Platelets     -0.064403  0.143733 -0.120147       0.103212   1.000000   
Prothrombin    0.218224  0.089384  0.112174       0.020122  -0.167331   
Stage          0.269400  0.041273  0.164945       0.123899  -0.253523   

               Prothrombin     Stage  
ID               -0.191930 -0.033757  
N_Days           -0.111470 -0.366193  
Age               0.113760  0.189083  
Bilirubin         0.314894  0.200731  
Cholesterol      -0.030811  0.011164  
Albumin          -0.200592 -0.305296  
Copper            0.218224  0.269400  
Alk_Phos          0.089384  0.041273  
SGOT              0.112174  0.164945  
Tryglicerides     0.020122  0.123899  
Platelets        -0.167331 -0.253523  
Prothrombin       1.000000  0.207802  
Stage             0.207802  1.000000

In [101]:
print(df.duplicated())

0      False
1      False
2      False
3      False
4      False
       ...  
413    False
414    False
415    False
416    False
417    False
Length: 418, dtype: bool


In [102]:
df.T
#Transposes rows and columns.

0                1                2    \
ID                           1                2                3   
N_Days                     400             4500             1012   
Status                       D                C                D   
Drug           D-penicillamine  D-penicillamine  D-penicillamine   
Age                      21464            20617            25594   
Sex                          F                F                M   
Ascites                      Y                N                N   
Hepatomegaly                 Y                Y                N   
Spiders                      Y                Y                N   
Edema                        Y                N                S   
Bilirubin                 14.5              1.1              1.4   
Cholesterol              261.0            302.0            176.0   
Albumin                    2.6             4.14             3.48   
Copper                   156.0             54.0            210.0   
Alk_Phos                1718.0           7394.8            516.0   
SGOT                    137.95           113.52             96.1   
Tryglicerides            172.0             88.0             55.0   
Platelets                190.0            221.0            151.0   
Prothrombin               12.2             10.6             12.0   
Stage                      4.0              3.0              4.0   

                           3        4        5        6        7    \
ID                           4        5        6        7        8   
N_Days                    1925     1504     2503     1832     2466   
Status                       D       CL        D        C        D   
Drug           D-penicillamine  Placebo  Placebo  Placebo  Placebo   
Age                      19994    13918    24201    20284    19379   
Sex                          F        F        F        F        F   
Ascites                      N        N        N        N        N   
Hepatomegaly                 Y        Y        Y        Y        N   
Spiders                      Y        Y        N        N        N   
Edema                        S        N        N        N        N   
Bilirubin                  1.8      3.4      0.8      1.0      0.3   
Cholesterol              244.0    279.0    248.0    322.0    280.0   
Albumin                   2.54     3.53     3.98     4.09      4.0   
Copper                    64.0    143.0     50.0     52.0     52.0   
Alk_Phos                6121.8    671.0    944.0    824.0   4651.2   
SGOT                     60.63   113.15     93.0    60.45    28.38   
Tryglicerides             92.0     72.0     63.0    213.0    189.0   
Platelets                183.0    136.0      NaN    204.0    373.0   
Prothrombin               10.3     10.9     11.0      9.7     11.0   
Stage                      4.0      3.0      3.0      3.0      3.0   

                           8        9    ...    408    409    410    411  \
ID                           9       10  ...    409    410    411    412   
N_Days                    2400       51  ...   1067   1072   1119   1097   
Status                       D        D  ...      C      C      C      C   
Drug           D-penicillamine  Placebo  ...    NaN    NaN    NaN    NaN   
Age                      15526    25772  ...  15706  14245  18628  24472   
Sex                          F        F  ...      F      F      F      F   
Ascites                      N        Y  ...    NaN    NaN    NaN    NaN   
Hepatomegaly                 N        N  ...    NaN    NaN    NaN    NaN   
Spiders                      Y        Y  ...    NaN    NaN    NaN    NaN   
Edema                        N        Y  ...      N      N      N      N   
Bilirubin                  3.2     12.6  ...    0.7    1.5    0.6    1.0   
Cholesterol              562.0    200.0  ...    NaN    NaN    NaN    NaN   
Albumin                   3.08     2.74  ...   3.73   3.81   3.57   3.58   
Copper                    79.0    140.0  ...    NaN    NaN    N

In [103]:
df['Drug'].value_counts

<bound method IndexOpsMixin.value_counts of 0      D-penicillamine
1      D-penicillamine
2      D-penicillamine
3      D-penicillamine
4              Placebo
            ...       
413                NaN
414                NaN
415                NaN
416                NaN
417                NaN
Name: Drug, Length: 418, dtype: object>

In [104]:
df.drop('Age',inplace = True, axis = 1)
df.head()

ID  N_Days Status             Drug Sex Ascites Hepatomegaly Spiders Edema  \
0   1     400      D  D-penicillamine   F       Y            Y       Y     Y   
1   2    4500      C  D-penicillamine   F       N            Y       Y     N   
2   3    1012      D  D-penicillamine   M       N            N       N     S   
3   4    1925      D  D-penicillamine   F       N            Y       Y     S   
4   5    1504     CL          Placebo   F       N            Y       Y     N   

   Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos    SGOT  Tryglicerides  \
0       14.5        261.0     2.60   156.0    1718.0  137.95          172.0   
1        1.1        302.0     4.14    54.0    7394.8  113.52           88.0   
2        1.4        176.0     3.48   210.0     516.0   96.10           55.0   
3        1.8        244.0     2.54    64.0    6121.8   60.63           92.0   
4        3.4        279.0     3.53   143.0     671.0  113.15           72.0   

   Platelets  Prothrombin  Stage  
0      190.0         12.2    4.0  
1      221.0         10.6    3.0  
2      151.0         12.0    4.0  
3      183.0         10.3    4.0  
4      136.0         10.9    3.0

In [105]:
new_df1 = df.copy()
new_df1.tail()

ID  N_Days Status Drug Sex Ascites Hepatomegaly Spiders Edema  \
413  414     681      D  NaN   F     NaN          NaN     NaN     N   
414  415    1103      C  NaN   F     NaN          NaN     NaN     N   
415  416    1055      C  NaN   F     NaN          NaN     NaN     N   
416  417     691      C  NaN   F     NaN          NaN     NaN     N   
417  418     976      C  NaN   F     NaN          NaN     NaN     N   

     Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos  SGOT  Tryglicerides  \
413        1.2          NaN     2.96     NaN       NaN   NaN            NaN   
414        0.9          NaN     3.83     NaN       NaN   NaN            NaN   
415        1.6          NaN     3.42     NaN       NaN   NaN            NaN   
416        0.8          NaN     3.75     NaN       NaN   NaN            NaN   
417        0.7          NaN     3.29     NaN       NaN   NaN            NaN   

     Platelets  Prothrombin  Stage  
413      174.0         10.9    3.0  
414      180.0         11.2    4.0  
415      143.0          9.9    3.0  
416      269.0         10.4    3.0  
417      350.0         10.6    4.0

In [106]:
new_df1.fillna(99, inplace = True)
new_df1.tail()

ID  N_Days Status Drug Sex Ascites Hepatomegaly Spiders Edema  \
413  414     681      D   99   F      99           99      99     N   
414  415    1103      C   99   F      99           99      99     N   
415  416    1055      C   99   F      99           99      99     N   
416  417     691      C   99   F      99           99      99     N   
417  418     976      C   99   F      99           99      99     N   

     Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos  SGOT  Tryglicerides  \
413        1.2         99.0     2.96    99.0      99.0  99.0           99.0   
414        0.9         99.0     3.83    99.0      99.0  99.0           99.0   
415        1.6         99.0     3.42    99.0      99.0  99.0           99.0   
416        0.8         99.0     3.75    99.0      99.0  99.0           99.0   
417        0.7         99.0     3.29    99.0      99.0  99.0           99.0   

     Platelets  Prothrombin  Stage  
413      174.0         10.9    3.0  
414      180.0         11.2    4.0  
415      143.0          9.9    3.0  
416      269.0         10.4    3.0  
417      350.0         10.6    4.0

In [107]:
new_df1.isna().sum()

ID               0
N_Days           0
Status           0
Drug             0
Sex              0
Ascites          0
Hepatomegaly     0
Spiders          0
Edema            0
Bilirubin        0
Cholesterol      0
Albumin          0
Copper           0
Alk_Phos         0
SGOT             0
Tryglicerides    0
Platelets        0
Prothrombin      0
Stage            0
dtype: int64

In [108]:
new_df1['Status'] = pd.factorize(new_df1.Status)[0]
new_df1['Drug'] = pd.factorize(new_df1.Drug)[0]
new_df1['Sex'] = pd.factorize(new_df1.Sex)[0]
new_df1['Ascites'] = pd.factorize(new_df1.Ascites)[0]
new_df1['Hepatomegaly'] = pd.factorize(new_df1.Hepatomegaly)[0]
new_df1['Spiders'] = pd.factorize(new_df1.Spiders)[0]
new_df1['Edema'] = pd.factorize(new_df1.Edema)[0]
print(new_df1)
# 0 = female
# 1 = male

      ID  N_Days  Status  Drug  Sex  Ascites  Hepatomegaly  Spiders  Edema  \
0      1     400       0     0    0        0             0        0      0   
1      2    4500       1     0    0        1             0        0      1   
2      3    1012       0     0    1        1             1        1      2   
3      4    1925       0     0    0        1             0        0      2   
4      5    1504       2     1    0        1             0        0      1   
..   ...     ...     ...   ...  ...      ...           ...      ...    ...   
413  414     681       0     2    0        2             2        2      1   
414  415    1103       1     2    0        2             2        2      1   
415  416    1055       1     2    0        2             2        2      1   
416  417     691       1     2    0        2             2        2      1   
417  418     976       1     2    0        2             2        2      1   

     Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos    SGOT 

In [110]:
new_df1['Bilirubin'] = new_df1['Bilirubin'].astype(int)
new_df1['Cholesterol'] = new_df1['Cholesterol'].astype(int)
new_df1['Albumin'] = new_df1['Albumin'].astype(int)
new_df1['Copper'] = new_df1['Copper'].astype(int)
new_df1['Alk_Phos'] = new_df1['Alk_Phos'].astype(int)
new_df1['SGOT'] = new_df1['SGOT'].astype(int)
new_df1['Tryglicerides'] = new_df1['Tryglicerides'].astype(int)
new_df1['Platelets'] = new_df1['Platelets'].astype(int)
new_df1['prothrombin'] = new_df1['Prothrombin'].astype(int)
new_df1['Stage'] = new_df1['Stage'].astype(int)
new_df1.dtypes

ID                 int64
N_Days             int64
Status             int64
Drug               int64
Sex                int64
Ascites            int64
Hepatomegaly       int64
Spiders            int64
Edema              int64
Bilirubin          int64
Cholesterol        int64
Albumin            int64
Copper             int64
Alk_Phos           int64
SGOT               int64
Tryglicerides      int64
Platelets          int64
Prothrombin      float64
Stage              int64
prothrombin        int64
dtype: object

In [112]:
new_df1.Status.value_counts()
# 0= death
# 1= censored
# 2=CL


1    232
0    161
2     25
Name: Status, dtype: int64

In [113]:
new_df1.Sex.value_counts()
# 0 = female
# 1 = male

0    374
1     44
Name: Sex, dtype: int64

In [114]:
new_df1.Drug.value_counts()
# 0= D-penicilian
# 1=placebo
# 2=unrecorded

0    158
1    154
2    106
Name: Drug, dtype: int64

In [115]:
new_df1.Ascites.value_counts()
# 0= yes
# 1= no
# 2 =unrecorded

1    288
2    106
0     24
Name: Ascites, dtype: int64

In [116]:
new_df1.Hepatomegaly.value_counts()
# 0= no
# 1=yes
# 2=unrecorded

0    160
1    152
2    106
Name: Hepatomegaly, dtype: int64

In [117]:
new_df1.Spiders.value_counts()
# 0= unrecoded
# 1= yes
# 2=no

1    222
2    106
0     90
Name: Spiders, dtype: int64

In [118]:
new_df1.Edema.value_counts()
# 0= no
# 1=yes
# 2 = unrecorded

1    354
2     44
0     20
Name: Edema, dtype: int64

In [119]:
new_df1.max()
# max return a maximum value in the column

ID                 418.0
N_Days            4795.0
Status               2.0
Drug                 2.0
Sex                  1.0
Ascites              2.0
Hepatomegaly         2.0
Spiders              2.0
Edema                2.0
Bilirubin           28.0
Cholesterol       1775.0
Albumin              4.0
Copper             588.0
Alk_Phos         13862.0
SGOT               457.0
Tryglicerides      598.0
Platelets          721.0
Prothrombin         99.0
Stage               99.0
prothrombin         99.0
dtype: float64

## **Conclusion**

A total of 418 people participated in this study, of which 158 patients took d-penicillin and complained of edema.